flow/record/fieldtypes/__init__.py

```diff
@@ -247,7 +247,8 @@ def __new__(cls, *args, **kwargs):
                             return cls.strptime(re.sub(RE_STRIP_NANOSECS, "\\1", arg), "%Y-%m-%dT%H:%M:%S.%f%z")
                     return cls.strptime(arg, "%Y-%m-%dT%H:%M:%S%z")
                 else:
-                    return cls.fromisoformat(arg)
+                    # Sometimes nanoseconds need to be stripped
+                    return cls.fromisoformat(re.sub(RE_STRIP_NANOSECS, "", arg))
```

Comment:
Most of the time nanosecs are not present, since this code is quite hot maybe consider using a try-catch like with the timezone version? Also, in future versions of Python this is very likely to be fixed (although I have to confess I have not checked the roadmap, it just seems logical) So this might be a good reason to only apply this fix in case of a ValueError?

Reply:
I changed the code accordingly. Any further comments?

Reply:
No, looks good, thanks. But can you fix the linting issue please?